content/articles/introduction-to-subnetting/index.md

```diff
@@ -0,0 +1,205 @@
+Class D networks (Multicast) Utilized in the multicasting of applications. Multicasting is the process where a single source communicates with many receivers.
+Class E networks (Reserved): Most network implementations recognize this address as undefined.
+
+#### Public vs Private IP 
+
+Public addresses are the ones used to connect to the internet. With a public IP, you can organize your server (VPN, FTP, WEB, etc), and access any other devices from anywhere on the globe.
+Private IP addresses are limited. They cannot receive traffic from the internet. Used to work within the local network. 
+        Example:
+        
+        i.10.0.0.1 - 10.255.255.255 
+        
+        ii.ii. 172.16.0.0 - 172.32.255.255
+        
+        iii.iii. 192.168.0.0 - 192.168.255.255.
+
+#### IPV4 vs IPV6
+
+Tabular representation of the differences between both versions of the internet protocol.
+
+| IPV4                    |     IPV6             |
+| ----------------------- | -------------------- |
+| The oldest version of the internet protocol | The newest version of the internet protocol |
+| 32 bits logical address  |  128 bits logical address  |
+| Uses dot-decimal notations, less suitable for mobile |  Uses hexadecimal colon-separated notations, best suited for mobile. |
+| Internet protocol security is optional  |  Internet protocol security is mandatory  |
+| The latest packet size is 576 bytes | The least packet size is 1208 bytes |
+| Its security depends on applications | Has it’s own security protocol called IPSec | 
+| Networks configured  with DHCP | Has its auto-configuration capabilities |
+| It is a numeric address method | It is an alphanumeric address method |
+| Has 5 different classes from A - E | Allows an unlimited number of IP addresses to be stored. |
+
+### Subnet Mask
+
+This is a term to denote the number of bits given to the network and the number given to the host. It is the process of setting the host bits to 0s and setting the network bits to 1s.  Here is the formula to calculate subnet;
+
+        For Network = 2n where n = number of desired bits. 
+        
+        For Hosts = 2n+2
+        
+        Example  to calculate subnets and hosts:
+        
+        Borrow 2 bits  (I.e n = 2)
+        
+| S   | S   | H   | H   | H   | H   | H   | H   |
+| --- | --- | --- | --- | --- | --- | --- | --- |
+        Number(#) of subnets = 22 = 4. 
+        Subnet mask = 2bits = 128 + 64 = 192
+        Range of hosts = 26 = 64   
+
+|        | Range      | Usable Range |
+| ------ | ---------- | ------------ |
+| Network ID | 0-63 | 65 - 126 |
+|     |  64-127 | 129 - 190 |
+| Broadcast Address | 192 - 255 |     |
```

Comment:
The formula is meant to be 2^n + 2, isn't it? Double check the math in this section